denizen_scripts/survival/repo_link/temporary/world_event.dsc

```diff
@@ -0,0 +1,173 @@
+  type: inventory
+  debug: false
+  inventory: chest
+  title: <&a>World Event
+  definitions:
+    G: white_stained_glass_pane[display_name=<&a>]
+    T: "ender_chest[display_name=<&a>Turn in Materials;lore=<&e>- Iron Ingot|<&e>- Iron Block;nbt=action/turn_in]"
+    soon: "barrier[display_name=<&e>Coming Soon!]"
+  slots:
+    - [G] [G] [G] [G] [G] [G] [G] [G] [G]
+    - [G] [G] [G] [G] [G] [G] [G] [G] [G]
+    - [G] [] [] [] [G] [] [] [] [G]
+    - [G] [] [] [G] [G] [G] [] [] [G]
+    - [G] [G] [G] [T] [G] [G] [G] [G] [G]
+    - [G] [G] [G] [G] [G] [G] [G] [G] [G]
+
+world_event_progress_inventory_events:
+  type: world
+  debug: false
+  events:
+    on player clicks item in world_event_progress_inventory:
+      - determine passively cancelled
+      - if <context.item.has_nbt[action]> && <context.item.nbt[action]> == turn_in:
+        - inventory open d:world_event_turn_in
+
+world_event_progress_inventory_open:
+  type: task
+  debug: false
+  script:
+    - define inventory <inventory[world_event_progress_inventory]>
+    - define percentage <server.flag[world_event.progress]./[<script[world_event_config].data_key[server_amount_needed]>]||0>
+    - if <[percentage]> > 1:
+      - define percentage 1
+    #- inventory set slot:5 o:world_event_bar_<[percentage].round_to_precision[0.2].*[5]> d:<[inventory]>
+    - define "lore:|:<&e>Server<&co> <&b><server.flag[world_event.progress]||0>/<script[world_event_config].data_key[server_amount_needed]>  <&6>(<server.flag[world_event.progress]./[<script[world_event_config].data_key[server_amount_needed]>].*[100].round_to[2]||0>%)"
+    - define lore:|:<list.pad_right[<[percentage].*[20].round_down||0>].with[<&a>⬛].pad_right[20].with[<&7>⬛].unseparated>
+    - define lore:|:<&a>
+    - define percentage <player.flag[world_event.progress]./[<script[world_event_config].data_key[personal_amount_needed]>]||0>
+    - if <[percentage]> > 1:
+      - define percentage 1
+    - define "lore:|:<&e>Personal<&co> <&b><player.flag[world_event.progress]||0>/<script[world_event_config].data_key[personal_amount_needed]> <&6>(<player.flag[world_event.progress]./[<script[world_event_config].data_key[personal_amount_needed]>].*[100].round_to[2]||0>%)"
+    - define lore:|:<list.pad_right[<[percentage].*[20].round_down||0>].with[<&a>⬛].pad_right[20].with[<&7>⬛].unseparated>
+    - inventory set slot:5 "o:player_head[skull_skin=<player.skull_skin>;lore=<[lore]>;display_name=<&a>World Event Progress]" d:<[inventory]>
+    - inventory set slot:42 "o:currency_display_item[display_name=<&6>Tokens<&co><&e> <player.flag[world_event.tokens.current]||0>;lore=<&7>Awarded at the end of the week]" d:<[inventory]>
+    - give "iron_ingot[display_name=<&a>Week 1;lore=<&e>- Iron Ingots|<&e>- Iron Blocks]" to:<[inventory]>
+    - repeat 9:
+      - give "barrier[display_name=<&a>Week <[value].+[1]>;lore=<&b>Release Date<&co>|<&e><time[2020/11/03_02:20:31:123_-07:00].add[<[value].*[7]>d].format[MM/dd/YYYY]>]" to:<[inventory]>
```

Comment:
instead of re-parsing this line nine times you might just parse it once and repeat it as a list
```suggestion
    - repeat 9:
      - give "barrier[display_name=<&a>Week <[value].+[1]>;lore=<&b>Release Date<&co>|<&e><time[2020/11/03_02:20:31:123_-07:00].add[<[value].*[7]>d].format[MM/dd/YYYY]>]" to:<[inventory]>
```
```suggestion
    - give "<item[barrier].with[display_name=<&a>Week <[value].+[1]>;lore=<&b>Release Date<&co>|<&e><time[2020/11/03_02:20:31:123_-07:00].add[<[value].*[7]>d].format[MM/dd/YYYY]>].repeat_as_list[9]>" to:<[inventory]>
```

Reply:
That doesn't work.